docs/release.md

```diff
@@ -0,0 +1,103 @@
+- https://github.com/rancher/kontainer-driver-metadata/pull/1027
+- https://github.com/rancher/kontainer-driver-metadata/pull/1084 
+
+### Prepare release notes for KDM (TBD)
+
+## Review PRs 
+
+There are a few common things to check while reviewing PRs for release. Most of this should already have been reviewed during the developer workflow. We do end up catching last minute bugs when reviewing PRs at release time so it's often a good idea to double check things.
+
+RKE1: (Reviewers: @snasovich @kinarashah @jiaqiluo)
+1. Kubernetes versions are correct according to the release plan
+2. Rancher server information is correct
+3. Default kubernetes version is correct 
+4. Only one kubernetes version per minor version is released
+5. Addon version information is correct
+6. Service options is correct if updated for a patch version
+
+RKE2/K3s: (Reviewers: @snasovich @rancher-max @kinarashah @jiaqiluo)
+1. Kubernetes versions are correct according to the release plan (no RC versions)
+2. Rancher server information is correct
+3. Default kubernetes version is correct
+4. Server and Agent arg information is correct
+5. Chart information is correct
+6. Only one kubernetes version per minor version is released
+
+## Prepare draft release notes for RKE1
+
+1. Draft a release note under https://github.com/rancher/rke/releases, it should have no tag attached and title would be "Draft Release vX.Y.Z" 
+2. Copy paste the existing release notes from the latest tag 
+3. Add major bug fixes and enhancements based on PRs present in diff between the last release and latest HEAD commit. Contact developers of PRs if more context is required. If the diff only includes KDM data, update enhancement section with the latest k8s versions about to be release. 
+4. List of kubernetes versions to be released is populated automatically on our RC tags under `RKE Kubernetes versions` (Example: https://github.com/rancher/rke/releases/tag/v1.4.4-rc7)
+
+Note: This is our current process, but we also have https://github.com/rancherlabs/release-notes/tree/main/rke/ (which seems outdated at the moment). Skip this section if docs team is creating draft release notes.
+
+## Release KDM
+
+1. PRs to be merged by @kinarashah or @snasovich 
+2. QA to perform post release checks for KDM after drone publish tasks are successfully completed (monitor at https://drone-publish.rancher.io/rancher/kontainer-driver-metadata)
+3. Release RKE1 (refer to Release RKE1 section)
+4. QA to perform post release checks for RKE after RKE tags are available 
+5. Announce in [rancher forums](https://forums.rancher.com/c/announcements/) and slack. Example forums posts for reference: 
+- https://forums.rancher.com/t/kubernetes-v1-24-10-and-v1-23-16/40218
+- https://forums.rancher.com/t/kubernetes-v1-24-8-v1-23-14-and-v1-22-16/39559
+
+## Release RKE1
+
+We release RKE after KDM is released. Currently we are managing releases for 
+- RKE v1.3.x (`release/v1.3`)
```

Comment:
It might be helpful for those unfamiliar with the RKE branching strategy to specify that `release/v1.3` is for rancher 2.6, and `release/v1.4` is for rancher 2.7